src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -930,8 +932,19 @@ void primitive_inst::do_runtime_skip_gather() {
     auto idx_shape = _impl_params->get_input_layout(1).get_shape();
     auto idx_rank = idx_shape.size();
 
-    if (idx_rank > 1) {
-        GPU_DEBUG_TRACE_DETAIL << "-- Cannot optimize becuase of its indices rank " << idx_shape.size() << std::endl;
+    // If all input layout count is zero, then it can be optimized because all input will be skipped for empty output.
+    if (_impl_params->get_input_layout(0).count() == 0 && _impl_params->get_input_layout(1).count() == 0) {
```

Comment:
Wouldn't be better to check output layout elements count? I think this condition doesn't help for cases when only one input is empty.

Reply:
Yeah that make sense

Reply:
Fixed it. thanks.